internal/scheduler/scheduler.go

```diff
@@ -165,6 +173,19 @@ func (s *Scheduler) Start(ctx context.Context, wg *sync.WaitGroup) error {
 	return nil
 }
 
+// RunNow attempts to trigger the scheduling loop, if the scheduling loop is actively running it will
+// cause the loop to run again immediately after finishing.
+func (s *Scheduler) RunNow() {
+	s.runNowRequested = time.Now()
+
+	// Attempt to wake up scheduling loop early, if it is already running do not block
+	// it will run again immediately after based on runNowRequested time.
+	select {
+	case s.runNow <- struct{}{}:
+	default:
+	}
+}
+
 func (s *Scheduler) start(ctx context.Context) {
 	const op = "scheduler.(Scheduler).start"
 	timer := time.NewTimer(s.runJobsInterval)
```

Comment:
I wonder if we should wake up the scheduler immediately when it starts instead of waiting for the full interval before running it for the first time.  Since the scheduler has just started up we don't know the last time any of the jobs were run.  Also running the jobs immediately at start up might make testing easier since our job can notify us when the scheduler has woken it up for the first time instead of having to add an arbitrary sleep in our tests, which end up being a flaky solution anyways.

Reply:
Yeah this had crossed my mind as well, ill make this change